recipes/default.rb

```diff
@@ -1,13 +1,79 @@
-package value_for_platform(
-  ['centos', 'redhat'] => { 'default' => 'tor-core' },
-  ['debian', 'ubuntu'] => { 'default' => 'tor' }
-)
+#
+# Author:: Richard Klafter (<rpklafter@yahoo.com>)
+# Cookbook Name:: tor
+# Recipe:: default
+#
 
+case node['platform_family']
+# Debian / Ubuntu
+when 'debian'
+  include_recipe 'apt'
+
+  # Add TorProject.org repository
+  apt_repository 'tor' do
+    uri          'http://deb.torproject.org/torproject.org'
+    distribution node['lsb']['codename']
+    components   ['main']
+    keyserver    'keys.gnupg.net'
+    key          '886DDD89'
+    deb_src      true
+  end
+
+  # Install Tor
+  package 'tor'
+
+# RHEL / Fedora
+when 'rhel', 'fedora'
+  include_recipe 'yum'
+
+  # Add TorProject.org repository
+  platformShort = node['platform_family'] == 'rhel' ? 'el' : 'fc'
+  intVersion = node['platform_version'].to_i
+  yum_repository 'tor' do
+    description "Tor Stable repo"
+    baseurl "https://deb.torproject.org/torproject.org/rpm/#{platformShort}/#{intVersion}/$basearch/"
+    gpgkey 'https://deb.torproject.org/torproject.org/rpm/RPM-GPG-KEY-torproject.org.asc'
+    action :create
+  end
+  yum_repository 'tor-source' do
+    description "Tor Source repo"
+    baseurl "https://deb.torproject.org/torproject.org/rpm/#{platformShort}/#{intVersion}/SRPMS/"
+    gpgkey 'https://deb.torproject.org/torproject.org/rpm/RPM-GPG-KEY-torproject.org.asc'
+    action :create
+  end
+
+  # Exclude platform Tor package
+  if node['platform_family'] == 'rhel' then
+    yum_repository 'epel' do
+      description 'Extra Packages for Enterprise Linux'
+      mirrorlist 'https://mirrors.fedoraproject.org/mirrorlist?repo=epel-6&arch=$basearch'
+      gpgkey 'http://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-6'
```

Comment:
`http://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-6` - > `https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-6` Do not grab keys over HTTP